src/components/SchemaViewer.svelte

```diff
@@ -11,15 +11,29 @@
       .split(" ")
       .filter((t) => t.length > 0);
 
-    const addVisibility = (node) => {
+    const addVisibility = (node, parentNodeNames = ["__root__"]) => {
       let modifiedNode = node;
+      let parentNames = [...parentNodeNames];
 
       if (modifiedNode.fields) {
-        modifiedNode.fields.forEach(addVisibility);
+        parentNames = modifiedNode.parentNames
+          ? `${modifiedNode.parentNames}.${modifiedNode.name}`
+          : modifiedNode.name;
+
+        modifiedNode.fields.forEach((field) => {
+          let modifiedNodeField = field;
+          modifiedNodeField.parentNames = parentNames;
```

Comment:
Remove this line and adjust accordingly. Treat the node as if it cannot have fields added to it on the fly.

```suggestion

```